checks/pinned_dependencies.go

```diff
@@ -786,3 +806,133 @@ func validatePackageManagerFile(name string, dl checker.DetailLogger, data FileC
+	// NOT module and frozen dependencies will set module to false and frozen to true
+	case (!isModule) && (frozen):
+		addPinnedResult(&w.module, false)
+		addPinnedResult(&w.frozen, true)
+	// default will set both flags to false
+	default:
+		addPinnedResult(&w.module, false)
+		addPinnedResult(&w.frozen, false)
+	}
+}
+
+func testIsDependencyFrozen(content []byte) (int, error) {
+	var r frozenPinningResult
+	_, err := validateIsDependencyFrozenIsPinned(content, &r)
+	return createReturnForIsDepedencyFrozen(r, err)
+}
+
+func createReturnForIsDepedencyFrozen(r frozenPinningResult, err error) (int, error) {
+	return createReturnValuesForIsDependencyFrozen(r, err)
+}
+
+func createReturnValuesForIsDependencyFrozen(r frozenPinningResult, err error) (int, error) {
+	if err != nil {
+		return checker.InconclusiveResultScore, err
+	}
+
+	switch {
+	case (r.module == pinned) && (r.frozen == pinned):
+		return checker.MinResultScore, nil
+	case (r.module == pinned) && (r.frozen == notPinned):
+		return checker.MaxResultConfidence, nil
+	case (r.module == notPinned) && (r.frozen == pinned):
+		return checker.MinResultScore, nil
+	case (r.module == notPinned) && (r.frozen == notPinned):
+		return checker.MaxResultConfidence, nil
+	default:
+		return checker.MinResultScore, nil
+	}
+}
+
+func isDependenciesFrozen(c *checker.CheckRequest) (int, error) {
+	var r frozenPinningResult
+	err := CheckFilesContent("package.json", false, c, validateDependencyPackageManager, &r)
+
+	if (err == nil) && (r.module == pinned) {
+		err = CheckIfFileExists(CheckPinnedDependencies, c, validateShrinkWrapJSON, &r)
+	}
+
+	return createReturnForIsDepedencyFrozen(r, err)
+}
+
+func validateShrinkWrapJSON(name string, dl checker.DetailLogger, data FileCbData) (bool, error) {
+	pdata, ok := data.(*frozenPinningResult)
+
+	if !ok {
+		panic("type need to be of frozenPinningResult")
+	}
+
+	switch strings.ToLower(name) {
+	case "npm-shrinkwrap.json":
+		dl.Info("javascript lock file detected: %s", name)
+	default:
+		return true, nil
+	}
+
+	addPinnedResult(&pdata.frozen, true)
+	return false, nil
+}
+
+func validateDependencyPackageManager(pathfn string, content []byte,
+	dl checker.DetailLogger, data FileCbData) (bool, error) {
+	switch pathfn {
+	case "package.json":
+		return validateIsDependencyFrozenIsPinned(content, data)
+	default:
+		return true, nil
+	}
+}
+
+func validateIsDependencyFrozenIsPinned(content []byte, data FileCbData) (bool, error) {
+	pdata, ok := data.(*frozenPinningResult)
+	if !ok {
+		// panic if it is not correct type.
+		panic("type need to be of frozenPinningResult")
+	}
+
+	pjson := PackageJSON{}
+	err := json.Unmarshal(content, &pjson)
+	if err != nil {
+		return false, sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("%v: %v", errInternalInvalidYamlFile, err))
+	}
+
+	// use regex to check no dependencies are pinned
+	// ^([0-9]+)\.([0-9]+)\.([0-9]+)?$
+	regex := regexp.MustCompile(`^([0-9]+)\.([0-9]+)\.([0-9]+)?$`)
```

Comment:
Is it enough to check fo the presence of `package-lock.json` and `npm-shrinkwrap.json`? My understanding is that we don't have to parse the `package.json` because it's changed at compilation time, whereas `package-lock.json` and `npm-shrinkwrap.json` don't and are the source of truth `package.json` uses.